netboxlabs/diode/sdk/client.py

```diff
@@ -234,6 +255,52 @@ def _setup_sentry(
         sentry_sdk.set_tag("platform", self._platform)
         sentry_sdk.set_tag("python_version", self._python_version)
 
+    def _authenticate(self):
+        authentication_client = _DiodeAuthentication(self._target, self._tls_verify, self._client_id, self._client_secret, self._path)
+        access_token = authentication_client.authenticate()
+        self._metadata = list(filter(lambda x: x[0] != "authorization", self._metadata)) + \
+            [("authorization", f"Bearer {access_token}")]
+
+
+class _DiodeAuthentication:
+    def __init__(self, target: str, path: str, tls_verify: bool, client_id: str, client_secret: str):
+        self._target = target
+        self._tls_verify = tls_verify
+        self._client_id = client_id
+        self._client_secret = client_secret
+        self._path = path
+
+    def authenticate(self) -> str:
+        """Request an OAuth2 token using client credentials and return it."""
+        if self._tls_verify:
+            conn = http.client.HTTPSConnection(
+                self._target,
+                context=None if self._tls_verify else ssl._create_unverified_context(),
+            )
+        else:
+            conn = http.client.HTTPConnection(
+                self._target,
+            )
+        headers = {"Content-type": "application/x-www-form-urlencoded"}
+        data = urlencode(
+            {
+                "grant_type": "client_credentials",
+                "client_id": self._client_id,
+                "client_secret": self._client_secret,
+            }
+        )
+        conn.request("POST", f"{self._path}/auth/token", data, headers)
```

Comment:
There's a few potential issues with that f-string to auth token when self._path is:
- a string with `/` resulting with `//auth/token`
- a string with tailing forward slash `foo/` resulting with `foo//auth/token`